pkg/sink/pulsar/config.go

```diff
@@ -0,0 +1,150 @@
+package pulsar
+
+import (
+	"errors"
+	"github.com/apache/pulsar-client-go/pulsar/log"
+	"github.com/sirupsen/logrus"
+	"time"
+
+	"github.com/apache/pulsar-client-go/pulsar"
+)
+
+type pulsarConfig struct {
+	// Configure the service URL for the Pulsar service. If you have multiple brokers, you can set multiple Pulsar cluster addresses for a client.This parameter is required.
+	URL   string `yaml:"url,omitempty" validate:"required"`
+	Topic string `yaml:"topic,omitempty" validate:"required"`
+	// Set the operation timeout. Producer-create, subscribe and unsubscribe operations will be retried until this interval, after which the operation will be marked as failed
+	OperationTimeoutSeconds    time.Duration `yaml:"operation_timeout_seconds,omitempty"`
+	UseTLS                     bool          `yaml:"useTLS,omitempty"`
+	TLSTrustCertsFilePath      string        `yaml:"tlsTrustCertsFilePath,omitempty"`
+	TLSAllowInsecureConnection bool          `yaml:"tlsAllowInsecureConnection,omitempty"`
+	// Max number of connections to a single broker that is kept in the pool. default 1
+	MaxConnectionsPerBroker int    `yaml:"maxConnectionPerBroker,omitempty"`
+	LogLevel                string `yaml:"logLevel,omitempty"`
+
+	CertificatePath string `yaml:"certificatePath,omitempty"`
+	PrivateKeyPath  string `yaml:"privateKeyPath,omitempty"`
+	Token           string `yaml:"token,omitempty"`
+	TokenFilePath   string `yaml:"tokenFilePath,omitempty"`
+	// Timeout for the establishment of a TCP connection
+	ConnectionTimeout time.Duration `yaml:"connectionTimeout,omitempty"`
+
+	// producer name
+	Name       string            `yaml:"name,omitempty"`
+	Properties map[string]string `yaml:"properties,omitempty"`
+	// SendTimeout set the timeout for a message that is not acknowledged by the server 30s.default 30s
+	SendTimeout time.Duration `yaml:"sendTimeout,omitempty"`
+	// MaxPendingMessages set the max size of the queue holding the messages pending to receive an acknowledgment from the broker.
+	MaxPendingMessages      int                    `yaml:"maxPendingMessages,omitempty"`
+	HashingScheme           pulsar.HashingScheme   `yaml:"hashingSchema,omitempty"`
+	CompressionType         pulsar.CompressionType `yaml:"compressionType,omitempty"`
+	Batching                bool                   `yaml:"batching,omitempty"`
+	BatchingMaxPublishDelay time.Duration          `yaml:"batchingMaxPublishDelay,omitempty"`
+	BatchingMaxMessages     uint                   `yaml:"batchingMaxMessages,omitempty"`
```

Comment:
Think about default 2048 as loggie batch size

Reply:
ok, this will be set